### PR TITLE
[Feature] Auto-DoubleToFloat

### LINKED ISSUE #1437: [BUG] DoubleToFloat transform deletes keys
## Describe the bug

The `DoubleToFloat` transform appears to delete all input keys, and will break any environment it is attached to.

## To Reproduce

Steps to reproduce the behavior.

```python
from torchrl.envs import DoubleToFloat, TransformedEnv, Compose
from torchrl.envs.libs.gym import GymEnv
GymEnv("CartPole-v0").reset()
```
```python
TensorDict(
    fields={
        done: Tensor(shape=torch.Size([1]), device=cpu, dtype=torch.bool, is_shared=False),
        observation: Tensor(shape=torch.Size([4]), device=cpu, dtype=torch.float32, is_shared=False)},
    batch_size=torch.Size([]),
    device=cpu,
    is_shared=False)
```
```python
TransformedEnv(GymEnv("CartPole-v0"), Compose(DoubleToFloat())).reset()
```
```python
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/smorad/code/from_src/torchrl/torchrl/envs/common.py", line 944, in reset
    tensordict_reset = self._reset(tensordict, **kwargs)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/smorad/code/from_src/torchrl/torchrl/envs/transforms/transforms.py", line 655, in _reset
    out_tensordict = self.transform._call(out_tensordict)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/smorad/code/from_src/torchrl/torchrl/envs/transforms/transforms.py", line 836, in _call
    tensordict = t._call(tensordict)
                 ^^^^^^^^^^^^^^^^^^^
  File "/Users/smorad/code/from_src/torchrl/torchrl/envs/transforms/transforms.py", line 201, in _call
    for in_key, out_key in zip(self.in_keys, self.out_keys):
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: 'NoneType' object is not iterable
```

## Expected behavior

Not crash

## System info

master, commit d500488f4e75d482792251725125568751360ccd

## Reason and Possible fixes

If you know or suspect the reason for this bug, paste the code lines and suggest modifications.

## Checklist

- [x] I have checked that there is no similar issue in the repo (**required**)
- [x] I have read the [documentation](https://github.com/pytorch/rl/tree/main/docs/) (**required**)
- [x] I have provided a minimal working example to reproduce the bug (**required**)

### PR DESCRIPTION
**Pull request by vmoens**
## Description

Closes #1437 

As per the discussion below, we also introduce a ``dTypeCastTransform``.
This class is tested through Double2Float: we do this as transform tests take a long time to run and double2float tests cover all functionalities of the parent class. This is not optimal but it saves about a minute of CI runtime.

### CODE REVIEW

**Comment by vmoens** [comment]
Cc @smorad 
Wdyt?

**Comment by smorad** [comment]
LGTM. It might be worth allowing a user-specified `dtype` in case the user wants half precision or something, but I think it's fine either way. At that point maybe it shouldn't be called `DoubleToFloat` anymore 😛

**Comment by vmoens** [comment]
Since this is expected to be more common we can either 
- set these dtypes as default and rename the class
- create a super class and instantiate this as a special case

**Comment by smorad** [comment]
IDK if it's common, but I can imagine some people wanting to use half-precision on something like Atari. I think approach 1 makes more sense.

**Comment by vmoens** [comment]
But it's bc breaking 😜
I can do 2 and put a warning

**Comment by vmoens** [comment]
@smorad Can you stamp this?

File: torchrl/envs/transforms/transforms.py [hunk 1]
```diff
@@ -2583,6 +2589,145 @@ def __repr__(self) -> str:
...
+            if self._keys_inv_unset:
+                in_keys_inv = []
+                state_spec = env_base.state_spec
+                if state_spec is not None:
+                    for key, spec in state_spec.items(True, True):
+                        if spec.dtype == torch.float64:
+                            in_keys_inv.append(unravel_key(key))
+                action_spec = env_base.action_spec
+                if action_spec.dtype == torch.float64:
+                    in_keys_inv.append(unravel_key(env_base.action_key))
```
Comment (smorad) [inline torchrl/envs/transforms/transforms.py:2725]: Is this a necessary override? It looks like `dTypeCastTransform` already implements `_set_in_keys` in a more general manner.

**Comment by vmoens** [comment]
Addressed comments from @smorad

**Pull request merged**